### ISSUE TITLE
RKE2 SELinux Support

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rke2#4]
This tracks any work that may be required to get SELinux support into RKE2.

**Comment by davidnuzik** [comment]
Bumping out of alpha milestone into beta.

**Comment by Oats87** [comment]
https://github.com/rancher/rke2-selinux is likely a relevant repository to this work

**Comment by davidnuzik** [comment]
Going forward we need to try to get better about linking PRs. I don't see any. This is also in "Working" but the work is done (should be in To Test).

**Comment by davidnuzik** [comment]
https://github.com/rancher/containerd/pull/4

Small stuff not directly related:
https://github.com/rancher/containerd/pull/5 (vendor rancher/cri + selinux)
https://github.com/rancher/rke2/pull/132 (bump version of containerd runtime)
https://github.com/rancher/rke2/pull/111 (fips compliant binaries)

I think that's pretty much it.

**Comment by davidnuzik** [comment]
Waiting on https://github.com/rancher/rke2/issues/156 until we can close this parent issue.

**Comment by davidnuzik** [comment]
I'm taking this out of To Test for now as we're blocked by https://github.com/rancher/rke2/issues/156 once this is done we can work on this parent issue. I just don't want this issue to show up as To Test if QA is actually not ready to work on this.

**Comment by ShylajaDevadiga** [comment]
Validating using rke2 beta17, on a multi node cluster
```
rke2 -v
rke2 version v1.18.4-beta17+rke2 (4c90fcab1e478d957d59e7ac9e4ebc1083330b79)

cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 7.8 (Maipo)

kubectl get nodes 
NAME                                          STATUS   ROLES         AGE     VERSION
ip-172-31-1-210.us-east-2.compute.internal    Ready    <none>        90s     v1.18.4-beta17+rke2
ip-172-31-14-121.us-east-2.compute.internal   Ready    etcd,master   14m     v1.18.4-beta17+rke2
ip-172-31-4-228.us-east-2.compute.internal    Ready    etcd,master   6m12s   v1.18.4-beta17+rke2
ip-172-31-6-138.us-east-2.compute.internal    Ready    etcd,master   3m27s   v1.18.4-beta17+rke2

getenforce
Enforcing

cat /var/lib/rancher/rke2/agent/etc/containerd/config.toml |grep selinux
  enable_selinux = true

ps -eZ|grep controller
system_u:system_r:rke2_service_t:s0:c20,c324 17486 ? 00:00:00 kube-controller

ps -eZ|grep nginx
system_u:system_r:container_t:s0:c369,c841 12878 ? 00:00:02 nginx-ingress-c
system_u:system_r:container_t:s0:c369,c841 12935 ? 00:00:00 nginx
```

**Comment by ShylajaDevadiga** [comment]
Ingress and dns resolution fails due to networkpolicy

```
curl -H  "host:foo1.bar.com" http://3.14.129.74/name.html
<html>
<head><title>504 Gateway Time-out</title></head>
<body>
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx/1.17.8</center>
</body>
</html>
```
Ingress works by deleting networkpolicy.
```
kubectl delete networkpolicy --all -n kube-system && kubectl delete networkpolicy --all -n default

curl -H  "host:foo1.bar.com" http://3.14.129.74/name.html 
testk8s1-7rqkm
```
PR to fix networkpolicy issue https://github.com/rancher/rke2/pull/299

**Comment by davidnuzik** [comment]
#299 has been merged in and is ready for retesting if you want to try it again.

**Comment by ShylajaDevadiga** [comment]
Waiting on a fix with nginx-ingress-controller issue #308 seen while validating using master build
rke2 version v1.18.4-dev+7c82ee79

**Comment by ShylajaDevadiga** [comment]
Using v1.18.8-beta19+rke2 
Valiated SELinux as per testplan.
Conformance test pased.
Upgrade using SUC is tracked here #359
